### PR TITLE
[ChatStateLayer] Add test coverage for ConnectedUser

### PR DESCRIPTION
**Pull request by laevandus**
### 🔗 Issue Links

Related: [#728](https://github.com/GetStream/ios-issues-tracking/issues/728)

*Merges to feature/chat-state-layer*

### 🎯 Goal

Tests for ConnectedUser

### ☑️ Contributor Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] This change follows zero ⚠️ policy (required)
- [ ] This change should be manually QAed
- [ ] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (docusaurus, tutorial, CMS)

### CODE REVIEW

File: Sources/StreamChat/StateLayer/ConnectedUserState.swift [hunk 1]
```diff
@@ -17,6 +17,9 @@ import Foundation
             with: .init(
                 userDidChange: { [weak self] in self?.user = $0 })
         )
+        if let user = observer.userObserver.item {
+            self.user = user
+        }
```
Comment (laevandus) [inline Sources/StreamChat/StateLayer/ConnectedUserState.swift:22]: Planning to do a separate PR for cleaning up some of the initial value part. Since I recently made state properties lazy then now it is important to always check the observer's state and not trust the injected user value which is part of the `init` (we might already have changed the user before state has been loaded, I caught this while writing tests).

File: Sources/StreamChat/Workers/CurrentUserUpdater.swift [hunk 2]
```diff
@@ -66,28 +66,32 @@ class CurrentUserUpdater: Worker {
         currentUserId: UserId,
         completion: ((Error?) -> Void)? = nil
     ) {
-        database.write { (session) in
+        database.write({ session in
```
Comment (laevandus) [inline Sources/StreamChat/Workers/CurrentUserUpdater.swift:69]: My tests failed because I was relying on database to update before this function returns. But here we did not use DB write's completion handler.

File: Sources/StreamChat/StateLayer/ConnectedUserState.swift (hunk 1 above)
Comment (martinmitrevski) [reply Sources/StreamChat/StateLayer/ConnectedUserState.swift:22]: probably we should also check other places

**Review by martinmitrevski — APPROVED** [review]
LGTM ✅
